client/src/main/java/io/pravega/client/segment/impl/SegmentOutputStreamImpl.java

```diff
@@ -93,24 +93,32 @@
         private long eventNumber = 0;
         private final ReusableLatch connectionSetup = new ReusableLatch();
         @GuardedBy("lock")
-        private CompletableFuture<Void> emptyInflightFuture = null;
+        private CompletableFuture<Exception> emptyInflightFuture = null;
 
         /**
          * Returns a future that will complete successfully once all the inflight events are acked
          * and will fail if that is not possible for some reason. IE: because the connection
          * dropped, or the segment was sealed.
          */
-        private CompletableFuture<Void> getEmptyInflightFuture() {
+        private CompletableFuture<Exception> getEmptyInflightFuture() {
```

Comment:
I think I understand what you want to do here, but I do find it a bit weird that you have a future that completes normally with an exception. What in particular did not work well before, when you were using `completeExceptionally`?